### PR TITLE
Delete specialcases

### LINKED ISSUE #5: Consider removing extranous grammar
The grammar seems to go to quite long lengths to define some very specific special-case constructions. For example, it handles `use constant` in a special way, and it handles `push`, `join` and `unpack` as special functions, but not most of the others.

A lot of these things can likely be removed without upsetting our primary use-case of nvim syntax highlight. There's no reason for each of these special-cases to be handled in any way other than "the same as everything else in Perl".

### PR DESCRIPTION
**Pull request by leonerd**
The grammar used to define some weird specialcase parsing rules for

```perl
use parent ...
use constant ...
use feature ...
use if ...
Module->import(...)
use subs ...
push ...
join ...
reverse ...
unpack ...
bless ...
```

All of these are just specialcases of more general grammar rules, and are not needed. This PR removes them.

Also deleted is one grammar rule that wasn't being used, and some helper functions from `grammar.js` itself that weren't being invoked.

There were no tests of any of the removed features so I didn't have to delete any of those. I have added some general tests of `use` statements though.

One unrelated test changed output, but honestly at this stage the output was just about as wrong as it is now and I'm disinclined to look into why. We'll get around to tidying those up later.

### CODE REVIEW

**Comment by leonerd** [comment]
Fixes #5

File: grammar.js [hunk 1]
```diff
@@ -136,18 +135,10 @@ module.exports = grammar({
...
       $.package_statement,
 
-      $.use_no_statement,
-      $.use_no_if_statement,
-      $.bareword_import,
-      $.use_no_subs_statement,
-      $.use_no_feature_statement,
       $.use_no_version,
+      $.use_no_statement,
       $.require_statement,
```
Comment (rabbiveesh) [inline grammar.js:140]: we should nuke `require` also; that's an expression.
Or is that not in the scope of this PR?

**Review by rabbiveesh — APPROVED** [review]
Looking it over now, I believe the original author's intent was to special case certain functions in his Language Server implementation (that's my guess at least).
Everything looks good to me by and large, pending the question about if making `require` into an expression should be this PR or a seperate issue

File: grammar.js (hunk 1 above)
Comment (leonerd) [reply grammar.js:140]: Lets keep require for now. There's a whole subcategory of "unary named operators" that aren't handled yet, so we'll move it there once we have those.

**Pull request merged**